File: Spring Boot/lms/src/main/java/com/courzelo/lms/entities/Class.java
package com.courzelo.lms.entities;

import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@Document(collection = "Question")
public class Class {

    @Id
    private String id;
    private String name ;
    private Long capacity ;

    private List<Program> programs ;

}

File: AngularFront/src/app/features/prog/models/Class.ts
import { Program } from "./program";

export class Class {

    public static fromJson(json: any): Class {
        return new Class(
            json['id'],
            json['name'],
            json['capacity'],
            json['programs'],
        )

    }

    constructor(

        public id: number,
        public name: string,
        public capacity: number,
        public programs: Program,
    ) { }

}